src/features/user/DonorReceipt/microComponents/DonorDetails.tsx

```diff
@@ -0,0 +1,66 @@
+import type { AddressView, DonorView } from '../donorReceipt';
+import type { CountryCode } from '@planet-sdk/common';
+
+import { useMemo } from 'react';
+import { useTranslations } from 'next-intl';
+import styles from '../donationReceipt.module.scss';
+import { getFormattedAddress } from '../../../../utils/addressManagement';
+
+interface Props {
+  donor: DonorView | undefined;
+  address: AddressView | undefined;
+}
+
+const DonorDetails = ({ donor, address }: Props) => {
+  if (address === undefined || donor === undefined) return null;
```

Comment:
I suggest removing this condition, updating the types for donor and address to not allow undefined. We would not want to show just donation details/receipt actions and no donor details.

If donor or address are absent in the data, it should be treated as an error scenario, an appropriate error message should be shown (invalid data encountered), along with a way to get in touch to fix things if needed. 

Perhaps this logic could be kept within ReceiptDataSection/DonorReceiptLayout instead, and we could create a component to show an error message instead of the rest of ReceiptDataSection.

Reply:
Error handling is needed.